clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java

```diff
@@ -508,4 +509,30 @@ public CreatePartitionsResult createPartitions(Map<String, NewPartitions> newPar
     public abstract CreatePartitionsResult createPartitions(Map<String, NewPartitions> newPartitions,
                                                             CreatePartitionsOptions options);
 
+    /**
+     * Delete records from topic partitions starting at the provided offsets.
```

Comment:
nit: `starting at the offsets` may be a bit confusing to users, as it may be interpreted as "from given offset to leo" whereas we are actually deleting "from lso to given offset (exclusive)". The javadoc on the old client may be more clarifying:

```
.. whose offset is smaller than the given offset of the corresponding partition
```